clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -646,7 +646,14 @@ LogicalResult LowerFunction::buildFunctionEpilog(const LowerFunctionInfo &FI) {
         if (auto al = findAlloca(ret)) {
           rewriter.replaceAllUsesWith(al.getResult(), RVAddr);
           rewriter.eraseOp(al);
+          rewriter.setInsertionPoint(ret);
           rewriter.replaceOpWithNewOp<ReturnOp>(ret);
+
+          auto retInputs = ret.getInput();
+          if (retInputs.size() == 1)
```

Comment:
Thanks for checking this, but what happens if `size != 1`? Looks like this would lead to a silent miscompilation. In practice we don't have returns with more than one input, but you need to an assert here to make sure we catch that if we end up changing the number of values a function may return. Example:
```
assert(retInputs.size() == 1 && "current number of support inputs")
if (auto load = dyn_cast<LoadOp>(retInputs[0].getDefiningOp()))
   if (load.getResult().use_empty())
      rewriter.eraseOp(load);
```